tidy3d/components/geometry.py

```diff
@@ -511,6 +511,28 @@ def reflect_points(
 
         return points_new
 
+    @abstractmethod
+    def volume(self, bounds: Bound) -> float:
+        """Returns object's volume. If the object extends beyound bounds then
+        the computed volume may be only a rough approximation
+
+        Returns
+        -------
+        float
+            Volume.
+        """
+
+    @abstractmethod
+    def surface_area(self, bounds: Bound) -> float:
+        """Returns object's surface area. If the object extends beyound bounds
+        then the computed area may be only a rough approximation
+
+        Returns
+        -------
+        float
+            Surface area.
+        """
+
```

Comment:
A docstring for the `bounds` parameter is missing.

Also, I feel like this should be an optional parameter, i.e. `bounds: Bound = None`. However, to implement this would require a lot of small changes/branching in the separate functions. One approach could be something like, in the geometry class:

```
    def volume(self, bounds: Bound = None):
        """Returns object's volume with optional bounds"""

        if not bounds:
            bounds = self.bounds

        return self._volume(bounds)
        
    @abstractmethod
    def _volume(self, bounds: Bound) -> float:
        """Returns object's volume given bounds."""
```

And then rename all subsequent methods to `_volume`. Same with `surface`. What do you think?

Reply:
that looks like a nice solution to me. Initially I decided not to worry about that since so many objects are used with `td.inf` dimension which would produce infinite outcome, so it seemed like a good idea just not to allow to call these functions without bounds

Reply:
Oh I see. That does make sense. When you put it this way I think it may be fine to leave as is, maybe just add the description of why bounds is needed in the docstring.

Reply:
Another option is to set the default bounds to something like `((-LARGE_NUMBER, LARGE_NUMBER), (-LARGE_NUMBER, LARGE_NUMBER), (-LARGE_NUMBER, LARGE_NUMBER))`. Because I think in many cases what I proposed will not return inf if there's an infinite dimension and bounds are not specified, it will just error out (numpy errors if you try math operations with `np.inf`).

Reply:
One thing that might clean things up just a bit is defining a method in Geometry:
```
def bounds_intersect(self, bounds: Bound) -> Bound:
    """Find the bounds intersecting the object's bounds with some supplied bounds."""
    rmin = tuple(max(c, c_) for c, c_ in zip(self.bounds[0], bounds[0]))
    rmax = tuple(min(c, c_) for c, c_ in zip(self.bounds[1], bounds[1]))
    return rmin, rmax
```
that can find the "intersection" between the instance bounds and some supplied bounds. Optionally, `bounds` could default to LARGE_NUMBERs